omnigibson/utils/usd_utils.py

```diff
@@ -568,12 +748,291 @@ def get_world_pose_with_scale(cls, prim_path):
+        idx = self._idx[prim_path]
+
+        # Load the current targets.
+        if "dof_actuation_forces" not in self._read_cache:
+            self._read_cache["dof_actuation_forces"] = self._view.get_dof_actuation_forces()
+
+        # Update the target
+        self._read_cache["dof_actuation_forces"][idx][indices] = efforts
+
+        # Add this index to the write cache
+        self._write_idx_cache["dof_actuation_forces"].add(idx)
+
+    def get_position_orientation(self, prim_path):
+        if "root_transforms" not in self._read_cache:
+            self._read_cache["root_transforms"] = self._view.get_root_transforms()
+
+        idx = self._idx[prim_path]
+        pose = self._read_cache["root_transforms"][idx]
+        return pose[:3], pose[3:]
+
+    def get_linear_velocity(self, prim_path):
+        if "root_velocities" not in self._read_cache:
+            self._read_cache["root_velocities"] = self._view.get_root_velocities()
+
+        idx = self._idx[prim_path]
+        return self._read_cache["root_velocities"][idx][:3]
+
+    def get_angular_velocity(self, prim_path):
+        if "root_velocities" not in self._read_cache:
+            self._read_cache["root_velocities"] = self._view.get_root_velocities()
+
+        idx = self._idx[prim_path]
+        return self._read_cache["root_velocities"][idx][3:]
+
+    def get_relative_linear_velocity(self, prim_path):
+        orn = self.get_position_orientation(prim_path)[1]
+        linvel = self.get_linear_velocity(prim_path)
+        return T.quat2mat(orn).T @ linvel
+
+    def get_relative_angular_velocity(self, prim_path):
+        orn = self.get_position_orientation(prim_path)[1]
+        angvel = self.get_angular_velocity(prim_path)
+        return T.mat2euler(T.quat2mat(orn).T @ T.euler2mat(angvel))
+
+    def get_joint_positions(self, prim_path):
+        if "dof_positions" not in self._read_cache:
+            self._read_cache["dof_positions"] = self._view.get_dof_positions()
+
+        idx = self._idx[prim_path]
+        return self._read_cache["dof_positions"][idx]
+
+    def get_joint_velocities(self, prim_path):
+        if "dof_velocities" not in self._read_cache:
+            self._read_cache["dof_velocities"] = self._view.get_dof_velocities()
+
+        idx = self._idx[prim_path]
+        return self._read_cache["dof_velocities"][idx]
+
+    def get_joint_efforts(self, prim_path):
+        if "dof_projected_joint_forces" not in self._read_cache:
+            self._read_cache["dof_projected_joint_forces"] = self._view.get_dof_projected_joint_forces()
+
+        idx = self._idx[prim_path]
+        return self._read_cache["dof_projected_joint_forces"][idx]
+
+    def get_mass_matrix(self, prim_path):
+        if "mass_matrices" not in self._read_cache:
+            self._read_cache["mass_matrices"] = self._view.get_mass_matrices()
+
+        idx = self._idx[prim_path]
+        # TODO: Maybe do the shape correction here. physics_view.mass_matrix_shape has it.
```

Comment:
does this still need to be fixed?

Reply:
@cgokmen

Reply:
Let's remove this todo. I believe you are able to correctly use this already.